File: calculator.py
print('Welcome to my calculator program')


# Add a prompt icon for display purposes
def prompt(message):
    print(f"==> {message}")


# Define an invalid number
def invalid_number(number_str):
    try:
        int(number_str)
    except ValueError:
        return True

    return False


# Ask the user for the first number
prompt('What is the first number?')
num1 = input()

# Check validity of first number
while invalid_number(num1):
    prompt("Hmm... That doesn't look like a valid number")
    num1 = input()

# Ask the user for the second number
prompt('What is the second number?')
num2 = input()

# Check validity of second number
while invalid_number(num2):
    prompt("Hmm... That doesn't look like a valid number")
    num2 = input()

# Ask the user for the operation to be performed
prompt('What operation would you like to be performed?\n1)Add '
       '2) Subtract 3)Multiply 4) Divide 5')
operation = input()

# Check validity of operation selection
while invalid_number(operation):
    prompt("Hmm... That doesn't look like a valid operation")
    operation = input()

# Perform the calculation of the two numbers - change to case match
match operation:
    case '1':
        result = int(num1) + int(num2)
    case '2':
        result = int(num1) - int(num2)
    case '3':
        result = int(num1) * int(num2)
    case '4':
        result = int(num1) / int(num2)

# Display the result of the calculation
prompt(f"The answer is {result}!")

File: lesson2/calculator.js
const readline = require("readline-sync");

// create invalid number function
function invalidNumber(number) {
  return number.trimStart() === "" || Number.isNaN(Number(number));
}

// create a prompt symbol
function prompt(message) {
  console.log(`=> ${message}`);
}

prompt("Welcome to the calculator");

// ask the user for the first number
prompt("What is the first number?");
let num1 = readline.question();

// validate the first number
while (invalidNumber(num1)) {
  prompt("Hmmm. That doesn't look like a valid number!");
  num1 = readline.question();
}

// ask the user for the second number
prompt("What is the second number?");
let num2 = readline.question();

// validate the second number
while (invalidNumber(num2)) {
  prompt("Hmmm.  That doesn't look like a valid number!");
  num2 = readline.question();
}

// ask the user what operation to perform
prompt(
  "What operation would you like to perform?\n1)Add 2)Subtract 3)Multiply 4)Divide"
);
let operation = readline.question();

// validate the operation
while (!["1", "2", "3", "4"].includes(operation)) {
  prompt("Must choose 1, 2, 3, or 4!");
  operation = readline.question();
}

// perform the operation
let output;
switch (operation) {
  case "1":
    output = Number(num1) + Number(num2);
    break;
  case "2":
    output = Number(num1) - Number(num2);
    break;
  case "3":
    output = Number(num1) * Number(num2);
    break;
  case "4":
    output = Number(num1) / Number(num2);
}

// display output of operation
prompt(`The result of your calculation is: ${output}`);
